bin/varnishadm/varnishadm.c

```diff
@@ -338,51 +340,66 @@ interactive(int sock)
 	}
 }
 
-/*
- * No arguments given, simply pass bytes on stdin/stdout and CLI socket
- */
-static void v_noreturn_
-pass(int sock)
-{
-	struct pollfd fds[2];
-	char buf[1024];
-	int i;
-	ssize_t n;
-	int busy = 0;
+/*--------------------------------------------------------------------*/
 
-	fds[0].fd = sock;
-	fds[0].events = POLLIN;
-	fds[1].fd = 0;
-	fds[1].events = POLLIN;
-	while (1) {
-		i = poll(fds, 2, -1);
-		if (i == -1 && errno == EINTR) {
-			continue;
-		}
-		assert(i > 0);
-		if (fds[0].revents & POLLIN) {
-			(void)pass_answer(fds[0].fd, pass_script);
-			busy = 0;
-			if (fds[1].fd < 0)
-				RL_EXIT(0);
-		}
-		if (fds[1].revents & POLLIN || fds[1].revents & POLLHUP) {
-			n = read(fds[1].fd, buf, sizeof buf - 1);
-			if (n == 0) {
-				if (!busy)
-					RL_EXIT(0);
-				fds[1].fd = -1;
-			} else if (n < 0) {
-				RL_EXIT(0);
-			} else {
-				busy = 1;
-				buf[n] = '\0';
-				cli_write(sock, buf);
-			}
-		}
+static void v_matchproto_(cli_func_t)
+ask_mgt(struct cli *cli, const char * const *av, void *priv)
+{
+	int i, sock, status;
+
+	AN(priv);
+	sock = *((int*)priv);
+	AN(cli_buf);
+	VSB_clear(cli_buf);
+	for (i = 1; av[i] != NULL; i++) {
+		VSB_quote(cli_buf, av[i], strlen(av[i]), 0);
+		VSB_putc(cli_buf, ' ');
+	}
```

Comment:
I think now is a good time to tackle the lack of SP quoting and share the logic with the `do_args()` function.

This could mean a new `VSB_QUOTE_CLI` flag. I'll let you think about our options here. One could be to always write double-quoted arguments with such a flag, except of course when we have a here-document.